src/renderer/VectorSkin.ts

```diff
@@ -106,7 +108,9 @@ export default class VectorSkin extends Skin {
     // This means that one texture pixel will always be between 0.5x and 1x the size of one rendered pixel,
     // but never bigger than one rendered pixel--this prevents blurriness from blowing up the texture too much.
     const mipLevel = VectorSkin.mipLevelForScale(scale);
-    if (!this._mipmaps.has(mipLevel)) this._createMipmap(mipLevel);
+    if (!this._mipmaps.has(mipLevel)) {
+      this._mipmaps.set(mipLevel, this._createMipmap(mipLevel));
+    }
 
     return this._mipmaps.get(mipLevel) ?? null;
```

Comment:
```suggestion

    let mipmap = this._mipmaps.get(mipLevel);
    // mipmap can be null if it's an invalid size, so only try creating it if it's undefined
    if (typeof mipmap !== "undefined") return mipmap;
    mipmap = this._createMipmap(mipLevel);
    return mipmap;
```

This lets us get rid of the `?? null` (which shouldn't be necessary because we know we'll never get `undefined`)

Reply:
Gross to type-check undefined. `.has()` exists for a reason. Agree that `?? null` is unpleasant, however.

Surely as simple as:

```suggestion
    if (this._mipmaps.has(mipLevel)) {
      return this._mipmaps.get(mipLevel);
    } else {
      const mipmap = this._createMipmap(mipLevel);
      this._mipmaps.set(mipLevel, mipmap);
      return mipmap;
    }
```

Reply:
\*frown\* moreover can't we just...?

```suggestion
    if (!this._mipmaps.has(mipLevel)) {
      this._mipmaps.set(mipLevel, this._createMipmap(mipLevel));
    }

    return this._mipmaps.get(mipLevel)!;
```

Don't actually prefer this to the if-else logic above. In ideal world TypeScript would tell if-not-has-then-set -> get guarantees get is not undefined, but—

well, suppose not. this indicates it is ["non-null"](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator) when they truly mean non-nullish. no one suggested typescript doesn't sin... :dog_shrugging:

Reply:
but obviously typescript should *implicitly* identify `.get(mipLevel)` is `typeof this._createMipmap(...)`. Too bad.

Reply:
ah. well, probably need the `as WebGLTexture | null` for if-has-return-get, anyway. :dog_facepawpadding:

```suggestion
    if (this._mipmaps.has(mipLevel)) {
      return this._mipmaps.get(mipLevel, mipmap) as WebGLTexture | null;
    } else {
      const mipmap = this._createMipmap(mipLevel);
      this._mipmaps.set(mipLevel, mipmap);
      return mipmap;
    }
```

Truly

Reply:
Yeah, TypeScript doesn't guard on `.has`.